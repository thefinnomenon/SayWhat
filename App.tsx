import React, { useContext } from 'react';
import { ThemeContext } from 'styled-components/native';
import ThemeManager from './contexts/ManageThemeContext';
import { I18nextProvider, useTranslation } from 'react-i18next';
import i18 from './i18n';
import { useScreens } from 'react-native-screens';
import { createAppContainer } from 'react-navigation';
import MainStackNavigator from './src/navigators/MainStackNavigator';

const AppContainer = createAppContainer(MainStackNavigator);

const Wrapper = () => {
  const themeContext = useContext(ThemeContext);
  const { t } = useTranslation();

  return <AppContainer screenProps={{ theme: themeContext, t }} />;
};

const App = () => {
  useScreens();

  return (
    <ThemeManager>
      <I18nextProvider i18n={i18}>
        <Wrapper />
      </I18nextProvider>
    </ThemeManager>
  );
};

export default App;
