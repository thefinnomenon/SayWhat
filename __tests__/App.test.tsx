import 'react-native';
import React, { ReactNode } from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
// @ts-ignore: No types
import mockAsyncStorage from '@react-native-community/async-storage/jest/async-storage-mock';
import App from '../App';

jest.mock('@react-native-community/async-storage', () => mockAsyncStorage);

jest.mock('react-native-code-push', () => {
  const cp = (opts: {}) => (app: ReactNode) => app;
  Object.assign(cp, {
    InstallMode: {},
    CheckFrequency: {},
    SyncStatus: {},
    UpdateState: {},
    DeploymentStatus: {},
    DEFAULT_UPDATE_DIALOG: {},

    checkForUpdate: jest.fn(),
    codePushify: jest.fn(),
    getConfiguration: jest.fn(),
    getCurrentPackage: jest.fn(),
    getUpdateMetadata: jest.fn(),
    log: jest.fn(),
    notifyAppReady: jest.fn(),
    notifyApplicationReady: jest.fn(),
    sync: jest.fn(),
  });
  return cp;
});

jest.mock('@react-native-community/audio-toolkit', () => {
  return {
    Player: () => ({
      play: jest.fn(),
      stop: jest.fn(),
      destory: jest.fn(),
    }),
  };
});

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
