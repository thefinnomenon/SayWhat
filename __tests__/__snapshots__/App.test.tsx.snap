// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#eee",
        "flex": 1,
        "flexDirection": "column-reverse",
        "overflow": "hidden",
      }
    }
  >
    <RNSScreenContainer
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        active={1}
        onComponentRef={[Function]}
        pointerEvents="auto"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                ]
              }
            />
            <Image
              source={
                Object {
                  "testUri": "../../../assets/images/logo.png",
                }
              }
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 2,
                    "flexShrink": 1,
                    "marginBottom": "auto",
                    "marginLeft": "auto",
                    "marginRight": "auto",
                    "marginTop": "auto",
                    "resizeMode": "contain",
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignContent": "center",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "space-around",
                    "marginBottom": "auto",
                    "marginLeft": "auto",
                    "marginRight": "auto",
                    "marginTop": "auto",
                    "width": "80%",
                  },
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignContent": "center",
                    "backgroundColor": "#50b7ff",
                    "borderRadius": 10,
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  PLAY
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                  ]
                }
              />
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignContent": "center",
                    "backgroundColor": "#50b7ff",
                    "borderRadius": 10,
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  SETTINGS
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                ]
              }
            />
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#f8f8f8",
            "transform": Array [
              Object {
                "translateX": -750,
              },
            ],
          }
        }
      >
        <View
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "backgroundColor": "#f8f8f8",
              "borderBottomColor": "#e0e0e0",
              "borderBottomWidth": 1,
              "height": 64,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
